python/taichi/profiler/profiler_impl.py

```diff
@@ -0,0 +1,164 @@
+from taichi.core import ti_core as _ti_core
+from taichi.lang import impl
+
+import taichi as ti
+
+
+class StatisticalResult:
+    def __init__(self, name):
+        self.name = name
+        self.counter = 0
+        self.min_time = 0.0
+        self.max_time = 0.0
+        self.total_time = 0.0
+
+    def __lt__(self, other):
+        if (self.total_time < other.total_time):
+            return True
+        else:
+            return False
+
+    def insert_record(self, time):
+        if self.counter == 0:
+            self.min_time = time
+            self.max_time = time
+        self.counter += 1
+        self.total_time += time
+        self.min_time = min(self.min_time, time)
+        self.max_time = max(self.max_time, time)
+
+
+class Profiler:
```

Comment:
1. more accurate name: `KernelProfiler`
2. Make modes constants so that we don't repeatedly type 'count' 'trace' literals